### PR TITLE
internal/dag: Add informers for TCPRoutes & UDPRoutes

### PR DESCRIPTION
**Pull request by stevesloka**
Adds informers to add TCP/UDP Routes to the k8s cache to allow setting status
on them in a future PR.

Updates #3441 #3442

Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

**Comment by stevesloka** [comment]
Adding release-note tag since it changes RBAC permissions.

**Review by sunjayBhatia — APPROVED** [review]

**Review by skriss — APPROVED** [review]
LGTM once the merge confict is resolved.

**Pull request merged**
